shared/workspaceService.ts

```diff
@@ -455,3 +456,13 @@ function buildRelative(
     relative: parameters.relative
   };
 }
+
+export function buildSubproblemPvfs(
+  criteria: Record<string, IProblemCriterion>
+): Record<string, IPvf> {
+  const dataSources: Record<string, IProblemDataSource> = _(criteria)
+    .flatMap((criterion): IProblemDataSource[] => criterion.dataSources)
+    .keyBy('id')
+    .value();
+  return _.mapValues(dataSources, (dataSource): IPvf => dataSource.pvf);
+}
```

Comment:
I'm not sure we want to grab them from the problem. I would prefer grabbing them from the subproblem instead. Using legacy data structures is bound to give trouble later on.